TYPO3.Flow/Classes/TYPO3/Flow/Resource/Storage/FileSystemStorage.php

```diff
@@ -128,38 +128,43 @@ public function getStreamByResourcePath($relativePath)
     /**
      * Retrieve all Objects stored in this storage.
      *
-     * @return array<\TYPO3\Flow\Resource\Storage\Object>
+     * @param callable $callback Function called after each iteration
+     * @return \Generator<\TYPO3\Flow\Resource\Storage\Object>
      */
-    public function getObjects()
+    public function getObjects(callable $callback = null)
     {
-        $objects = array();
         foreach ($this->resourceManager->getCollectionsByStorage($this) as $collection) {
-            $objects = array_merge($objects, $this->getObjectsByCollection($collection));
+            return $this->getObjectsByCollection($collection, $callback);
```

Comment:
Maybe I do not understand generators well enough, but how can a `foreach` loop with a return statement work over all entries?